pkg/config/dynamic_env.go

```diff
@@ -0,0 +1,128 @@
+package config
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+)
+
+type structuredEnv struct {
+	RemoteName string
+	LocalName  string
+}
+
+type (
+	dynamic_env struct {
+		Vault *vaultEnvProvider `toml:"vault"`
+	}
+	vaultEnvProvider struct {
+		EnvVars           []string        `toml:"env_vars"`
+		StructuredEnvVars []structuredEnv `toml:"-"`
+	}
+)
+
+// Get a list of env variables with possible renames like so:
+// "some_env:RENAME_NAME"
+// And extract them all in a structured { remoteName: "some_env", local_name: "RENAME_NAME" }
+func getEnvVars(EnvVars []string) []structuredEnv {
+	result := make([]structuredEnv, len(EnvVars))
+
+	for i, envVar := range EnvVars {
+		// Find first : that's not escaped with \
```

Comment:
How about finding the last `:` and skip handling escape? because env var name can't contain colon.